### PR TITLE
codeintel: updating ConfigurationEditor to use Apollo

### PR DESCRIPTION
**Pull request by Numbers88s**
This PR updates ConfigurationEditor component as part of the Apollo client updates.

### CODE REVIEW

File: client/web/src/enterprise/codeintel/configuration/RepositoryConfiguration.tsx [hunk 1]
```diff
@@ -1,25 +1,19 @@
...
 
 export interface RepositoryConfigurationProps extends ThemeProps, TelemetryProps {
     repo: { id: string }
-    updateConfigurationForRepository: typeof defaultUpdateConfigurationForRepository
-    getConfigurationForRepository: typeof defaultGetConfigurationForRepository
-    getInferredConfigurationForRepository: typeof defaultGetInferredConfigurationForRepository
     indexingEnabled: boolean
     history: H.History
+    isLightTheme: boolean
+    telemetryService: TelemetryService
```
Comment (efritz) [inline client/web/src/enterprise/codeintel/configuration/RepositoryConfiguration.tsx:16]: `ThemeProps` and `TelemetryProps` should include these fields already, no?

File: client/web/src/enterprise/codeintel/configuration/ConfigurationEditor.tsx [hunk 2]
```diff
@@ -1,92 +1,51 @@
...
+import {
+    useInferredConfig,
+    useRepositoryConfig,
+    useUpdateConfigurationForRepository,
+} from './usePoliciesConfigurations'
 
 export interface ConfigurationEditorProps extends ThemeProps, TelemetryProps {
     repoId: string
+    isLightTheme: boolean
+    telemetryService: TelemetryService
```
Comment (efritz) [inline client/web/src/enterprise/codeintel/configuration/ConfigurationEditor.tsx:24]: same comment here as below (also not sure I missed any of these in the other PR)

**Review by efritz — CHANGES_REQUESTED** [review]
You're on 🔥 👏 !

The only weird behavior I saw (that I think doesn't match current behavior, or at least expected behavior) is that after saving the configuration the page state isn't updated properly.

It should be that the discard action should no longer be enabled, and navigating away from the page shouldn't prompt you to check your unsaved changes (you just saved them!).

**Comment by Numbers88s** [comment]
Ah interesting, ok I'll take a look at that. Thank you!!

File: client/web/src/enterprise/codeintel/configuration/RepositoryConfiguration.tsx (hunk 1 above)
Comment (Numbers88s) [reply client/web/src/enterprise/codeintel/configuration/RepositoryConfiguration.tsx]: Yeah it does 👍  I removed the dups then. I missed it coming from the extends.

File: client/web/src/enterprise/codeintel/configuration/ConfigurationEditor.tsx (hunk 2 above)
Comment (Numbers88s) [reply client/web/src/enterprise/codeintel/configuration/ConfigurationEditor.tsx]: Yeah it does 👍  I removed the dups then. I missed it coming from the extends.

**Comment by Numbers88s** [comment]
I was under the impression that Apollo would handle the cache a little better, but its all good, I have more control over it now. It's fixed. Let me know what you think ✌️

**Review by efritz — APPROVED** [review]

**Comment by efritz** [comment]
image

**Pull request merged**